### PR TITLE
[SPARK-11559] [MLlib] Make `runs` no effect in mllib.KMeans

### PR DESCRIPTION
**Pull request by yanboliang**
## What changes were proposed in this pull request?

We deprecated  `runs` of mllib.KMeans in Spark 1.6 (SPARK-11358). In 2.0, we will make it no effect (with warning messages). We did not remove `setRuns/getRuns` for better binary compatibility.
This PR change `runs` which are appeared at the public API. Usage inside of `KMeans.runAlgorithm()` will be resolved at #10806. 
## How was this patch tested?

Existing unit tests.

cc @jkbradley

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala [hunk 1]
```diff
@@ -109,35 +108,20 @@ class KMeans private (
   }
 
   /**
-   * :: Experimental ::
-   * Number of runs of the algorithm to execute in parallel.
+   * This function has no effect since Spark 2.0.0.
    */
   @Since("1.4.0")
-  @deprecated("Support for runs is deprecated. This param will have no effect in 2.0.0.", "1.6.0")
```
Comment (felixcheung) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala]: isn't it better to keep the `@deprecated`?

Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala]: ```@deprecated``` means it take effect but not recommend to use, but in this PR we will make it no effect.

**Comment by jkbradley** [comment]
I'll take a look

File: python/pyspark/mllib/clustering.py [hunk 2]
```diff
@@ -349,9 +347,6 @@ def train(cls, rdd, k, maxIterations=100, runs=1, initializationMode="k-means||"
           rather than using the random or k-means|| initializationModel.
           (default: None)
         """
-        if runs != 1:
-            warnings.warn(
```
Comment (jkbradley) [inline python/pyspark/mllib/clustering.py]: I'd keep a warning, though you can remove the word "deprecated"

**Comment by jkbradley** [comment]
* pyspark.ml.clustering.KMeans needs to be updated too (in the docstrings)
* pyspark.mllib.clustering.KMeansModel has a doc test example which needs to be updated

**Comment by jkbradley** [comment]
That's all.  Thanks for separating this out.

**Comment by jkbradley** [comment]
LGTM
Thanks!
Merging with master

**Pull request closed without merging**